### PR TITLE
Expose more performance apis

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR adds or exposes several new features for performance measurments:

- `PerformanceMetricsTag` for measuring performance of compose
- start/finish a span during perfomence transaction
- allow for a custom transaction name

Related internal request: pdnsEh-1Lf-p2

image

### CODE REVIEW

File: crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceTransactionRepository.kt [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
+     *
+     * @see io.sentry.SpanStatus
+     */
+    fun finishSpan(spanId: SpanId, spanHttpCode: Int, throwable: Throwable? = null) {
+        val span = spans[spanId]
+        if (span != null) {
+            throwable?.let { span.throwable = throwable }
+            span.status = SpanStatus.fromHttpStatusCode(spanHttpCode)
+        }
+        span?.finish()
```
Comment (ParaskP7) [inline crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceTransactionRepository.kt]: Suggestion (💡): Consider the below alternative, which seems to me a bit less complex, one less `if` to worry about:

```kotlin
        spans[spanId]?.let { span ->
            throwable?.let { span.throwable = throwable }
            span.status = SpanStatus.fromHttpStatusCode(spanHttpCode)
            span.finish()
        }
```

File: crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceMetricsTag.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.automattic.android.tracks.crashlogging.performance
+
+import androidx.compose.ui.Modifier
+import io.sentry.compose.SentryModifier.sentryTag
+
+object PerformanceMetricsTag {
+    @JvmStatic
+    fun Modifier.tagForPerformanceMetrics(tag: String) = this then sentryTag(tag)
```
Comment (ParaskP7) [inline crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceMetricsTag.kt:8]: Praise (❤️): TIL about the `then` infix function on a modifier!

**Review by ParaskP7 — APPROVED** [review]
👋 @wzieba !

I have reviewed (and also tested this PR), everything looks and work as expected, good job! 🌟 

FYI: Although I (think) understand why you are not adding testing instruction, me having a Sentry test account, just like you, makes it "easy" to test should PRs. Consider in the future to add some testing instructions, even if that's "optional". This will save me time and effort to try and figure out what to test myself, or whether I tested at least what the PR author wanted me to test.

PS: Let me know when [BPAndroid#2826](https://github.com/beeper/beeper-android/pull/2826) is ready.

-----

I have left one suggestion (💡) for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

**Comment by mattylase** [comment]
Just reiterating my support here - did a run through of the APIs on the consumer side and everything is looking good 👍

File: crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceTransactionRepository.kt (hunk 1 above)
Comment (wzieba) [reply crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/performance/PerformanceTransactionRepository.kt]: Thanks! Addressed in 0f805e7c755546e98857feab9ba6a99d7f3d7d1e

**Pull request merged**